lib/apiservers/engine/backends/volume.go

```diff
@@ -58,7 +105,43 @@ func (v *Volume) VolumeRm(name string) error {
 		if _, ok := err.(*storage.RemoveVolumeConflict); ok {
 			return derr.NewRequestConflictError(fmt.Errorf("Volume is in use"))
 		}
-		return derr.NewErrorWithStatusCode(fmt.Errorf("Server error form portlayer"), http.StatusInternalServerError)
+		return derr.NewErrorWithStatusCode(fmt.Errorf("Server error from portlayer: %s", err), http.StatusInternalServerError)
 	}
 	return nil
 }
+
+//Utility Functions
+
+func fillDockerVolumeModel(volume *models.VolumeResponse, labels map[string]string) *types.Volume {
+	dockerVol := types.Volume{
+		Driver:     volume.Driver,
+		Name:       volume.Name,
+		Labels:     labels,
+		Mountpoint: volume.SelfLink,
+	}
+	return &dockerVol
+}
+
+func translateInputsToPortlayerRequestModel(name, host, driverName string, opts, labels map[string]string) models.VolumeRequest {
+	model := models.VolumeRequest{
+		Driver:     driverName,
+		DriverArgs: opts,
+		Name:       name,
+		Store:      host,
```

Comment:
Write and all of those things would get passed into the DriverArgs section when being translated to the portlayer hander. Which will negotiate the data before making the VolumeCreate call. So my question is how do we want to handle those args in the personality? George has expressed that no value should be passed into the portlayer as blank. So I talked to him about listing "default" as the value of any required args that are not set by the user. Then those defaults would be specified at deployment time with vic-machine. If not set the handler would talk to the storage layer(in the portlayer) and a "No default set, operation not permitted" type error should be returned and conveyed to the Personality.